osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/PlayerArea.cs

```diff
@@ -42,6 +42,11 @@ public class PlayerArea : CompositeDrawable
         /// </summary>
         public readonly SpectatorPlayerClock SpectatorPlayerClock;
 
+        /// <summary>
+        /// The clock adjustments applied by the <see cref="Player"/> loaded in this area.
+        /// </summary>
+        public readonly AudioAdjustments ClockAdjustmentsFromMods = new AudioAdjustments();
```

Comment:
Also may be able to get by with only exposing as `IAggregateAudioAdjustment`, which would make the flow a bit easier to follow (you can be sure that the adjustments are read only if so).

Reply:
I originally had it that way but it's null for a while during player's load, and I just didn't want to deal with peppering code in just the right way to avoid a nullref when this is referenced too early...

At best I could add a remark comment here?

Reply:
Can expose publicly as `IAggregateAudioAdjustment` and leave the `AudioAdjustment` as `private`, should still be better I think.

Reply:
> Can expose publicly as `IAggregateAudioAdjustment`

Oh, I didn't realise `IAggregateAudioAdjustment` was readonly. Done.